app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -168,33 +180,132 @@ class PopupView : FrameLayout {
+                    popupFieldValue.visibility = View.GONE
+                    //save original colors
+                    val oldColors: ColorStateList = popupFieldLabel.textColors
+                    // here we assign and hold the values of the editable fields, entered by the user
+                    // in popupAttribute.tempValue
+                    popupFieldEditValue.doAfterTextChanged {
+                        if (popupFieldEditValue.hasFocus()) {
 
-                        if (validationError != null) {
-                            val fieldLabelWithValidationError = popupField.label + ": " + validationError.message
-                            popupFieldLabel.text = fieldLabelWithValidationError
-                            popupFieldLabel.setTextColor(Color.RED)
-                        } else {
-                            popupFieldLabel.text = popupField.label
-                            popupFieldLabel.setTextColor(oldColors)
+                            val validationError: ArcGISRuntimeException? = updateValue(
+                                popupField,
+                                popupFieldEditValue.text.toString()
+                            )
+                            if (validationError != null) {
+                                val fieldLabelWithValidationError =
+                                    popupField.label + ": " + validationError.message
+                                popupFieldLabel.text = fieldLabelWithValidationError
+                                popupFieldLabel.setTextColor(Color.RED)
+                            } else {
+                                popupFieldLabel.text = popupField.label
+                                popupFieldLabel.setTextColor(oldColors)
+                            }
                         }
                     }
                 }
             } else {
                 popupFieldEditValue.visibility = View.GONE
+                popupFieldCvdValues.visibility = View.GONE
                 popupFieldValue.visibility = View.VISIBLE
             }
         }
+
+        /**
+         * Sets up spinner for PopupFields that have a CodedValueDomain.
+         */
+        private fun setUpSpinner(codedValueDomain: CodedValueDomain, popupField: PopupField) {
+            val codedValuesNames = mutableListOf<String>()
+            codedValueDomain.codedValues.forEach { codedValue -> codedValuesNames.add(codedValue.name) }
+            popupFieldCvdValues.adapter = ArrayAdapter(
+                binding.root.context,
+                android.R.layout.simple_spinner_dropdown_item,
+                codedValuesNames
+            )
+            val codedValuePosition = codedValueDomain.codedValues.indexOfFirst { codedValue ->
+                codedValue.code == popupManager.getFieldValue(popupField)
+            }
+            // set the PopupField value as selected in the spinner
+            popupFieldCvdValues.setSelection(codedValuePosition)
```

Comment:
since `codedValuePosition` can be -1 from `indexOfFirst`, would the spinner throw an exception if you set the selection to -1?
if so you could use what Soren suggested with an if statement instead 
```
       val index = codedValues.indexOfFirst { codedValue -> codedValue.code == code }
            return if (index == -1) 0 else index 
```

Reply:
@eri9000 I tested it out and no the `setSelection()` method does not throw an exception with `-1` .
It shows the zeroth element as being selected.